Assets/Plugins/Editor/Uplift/UpdateTool/Updater.cs

```diff
@@ -0,0 +1,131 @@
+            DirectoryInfo assetsPath = new DirectoryInfo(Application.dataPath);
+            string destination = Path.Combine(assetsPath.Parent.FullName, unitypackageName);
+            
+            ServicePointManager.ServerCertificateValidationCallback = delegate { return true; };
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(url, destination);
+            }
+
+            AssetDatabase.ImportPackage(destination, false);
+        }
+
+        public static void CheckForUpdate(bool forceCheck = false)
+        {
+            if(forceCheck || ShouldCheck())
+            {
+                EditorApplication.update += EditorUpdate;
+    			updateCoroutine = CheckForUpdateRoutine();
+            }
+        }
+        
+        private static void EditorUpdate()
+        {
+            updateCoroutine.MoveNext();
+        }
+
+        private static bool ShouldCheck()
+        {
+            string lastCheck = EditorPrefs.GetString(lastUpdateCheckKey);
+            if(string.IsNullOrEmpty(lastCheck)) return true;
+            
+            try
+            {
+                TimeSpan timeFromLastCheck = DateTime.UtcNow - DateTime.ParseExact(lastCheck, dateFormat, provider);
+                return timeFromLastCheck > TimeSpan.FromDays(1);
+            }
+            catch(FormatException e)
+            {
+                Debug.LogErrorFormat("The date stored for the last check of Uplift ({0}) is not in the correct format:\n{1}", lastCheck, e);
+            }
+            return true;
+        }
+        
+        private static IEnumerator CheckForUpdateRoutine() {
+			IEnumerator e = GetReleasesJson();
+			while (e.MoveNext()) yield return e.Current;
+
+			string json = (string) e.Current;
+
+			if (json == null) {
+				Debug.LogError ("Unable to check for Uplift updates");
+			} else {
+				System.Object obj = MiniJSON.Json.Deserialize (json);
+				List<System.Object> releases = (List<System.Object>)obj;
+				foreach (Dictionary<string, object> release in releases)
+				{
+                    if(VersionParser.GreaterThan((string)release ["tag_name"], About.Version))
+                    {
+                        Debug.Log("There is a new version of Uplift available!");
+                        UpdatePopup popup = EditorWindow.GetWindow(typeof(UpdatePopup), true) as UpdatePopup;
+                        popup.SetInformations(
+                            (string)release["tag_name"],
+                            ExtractUnityPackageUrl(release),
+                            (string)release["body"]
+                        );
+                    }
+                    else
+                    {
+                        Debug.Log("No update for Uplift available");
+                    }
+				}
+			}
+			EditorApplication.update -= EditorUpdate;
+            EditorPrefs.SetString(
+                lastUpdateCheckKey,
+                DateTime.UtcNow.ToString(dateFormat, provider)
```

Comment:
why not store time in seconds?

Reply:
Because I'm not sure you can in C#. Or we would have to write our own method, something like
```c#
public static int convertDateTimeToSeconds(DateTime dateTimeToConvert)
    {
        int secsInAMin = 60;
        int secsInAnHour = 60 * secsInAMin;
        int secsInADay = 24 * secsInAnHour;
        double secsInAYear = (int)365.25 * secsInADay;

        int totalSeconds = (int)(dateTimeToConvert.Year * secsInAYear) + 
                       (dateTimeToConvert.DayOfYear * secsInADay) +
                       (dateTimeToConvert.Hour * secsInAnHour) +
                       (dateTimeToConvert.Minute * secsInAMin) + 
                       dateTimeToConvert.Second;

        return totalSeconds;
    }
```

which would me more cumbersome than anything imo. DateTime.Second, .Minute etc... only returns the current seconds, minutes... (whithin the [0-60] range for instance)

Reply:
https://msdn.microsoft.com/en-us/library/system.datetime.ticks(v=vs.110).aspx

System.DateTime.Now.Ticks * 10000;

Reply:
The issue is that EditorPrefs does not accept `long`. Anyways if your first comment is true, then we should not use the Preferences, so we will not face the issue. I'm not sure I see why storing it as an integer is better than a string though.

Reply:
I was just not a fond of date parsing. I see that you use `CultureInfo.InvariantCulture` so it should be fine.